### PR TITLE
Redux components

### PR DESCRIPTION
**Pull request by MasumaJaffery**
In this Milestone;
1. Add functionality of useDispatch and useSelector.
2. React Components Modification by Redux
3. Setup Store by Provider.
Thanks for the Time to Review my PR.
Happy Coding 🤓

### CODE REVIEW

File: src/components/BookList.jsx [hunk 1]
```diff
@@ -1,40 +1,35 @@
 /* eslint-disable linebreak-style */
+// BookList.js
+
 import React from 'react';
-import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+function BookList() {
+  const dispatch = useDispatch();
+  const books = useSelector((state) => state.books.books);
```
Comment (muneebulrehman) [inline src/components/BookList.jsx]: You are trying to access `.books` on `state.books`, however, the final state of the books is the `state.books`. This is making the `const books` undefined and leading to no books being rendered on the UI. Kindly fix this error. ⬇️
image

File: src/components/BookList.jsx [hunk 2]
```diff
@@ -1,40 +1,35 @@
...
-function BookList({ books, onRemove }) {
   // Check if books is undefined or empty before rendering
   if (!books || books.length === 0) {
     return <h5>No books available.</h5>;
   }
 
+  const handleRemoveBook = (bookId) => {
+    dispatch({ type: 'REMOVE_BOOK', payload: bookId });
+  };
+
```
Comment (muneebulrehman) [inline src/components/BookList.jsx]: You need to dispatch the `removeBook` reducer that you created in the **booksSlice.js** file after you import it in this file. Currently this functionality is not working. Exactly the way you did for adding the book. Kindly fix this issue.

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -2,7 +2,27 @@ import { createSlice } from '@reduxjs/toolkit';
...
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
```
Comment (muneebulrehman) [inline src/redux/books/booksSlice.js]: In your `removeBook` reducer, you are filtering the books on the basis of `id` but `id` prop doesn't exist on the books. Instead it is `item_id`. Kindly either change it in the data here or update your reducer and `handleRemoveBook` method.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @MasumaJaffery , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good job setting up the redux store and accessing data in the react components.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ]  You are displaying only the book name and not the author's name. It would be better if you would also display the author's name along side the book name. That is what you will have to do in the next milestone anyways. Consider implementing this requirement.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @MasumaJaffery   , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional Readme file 
- [x] No linter errors
- [x] Redux components added correctly


 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**